compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -576,12 +575,6 @@ fn get_body_span<'tcx>(
 
 fn hash_mir_source<'tcx>(tcx: TyCtxt<'tcx>, hir_body: &'tcx rustc_hir::Body<'tcx>) -> u64 {
     // FIXME(cjgillot) Stop hashing HIR manually here.
-    let mut hcx = tcx.create_no_span_stable_hashing_context();
-    let mut stable_hasher = StableHasher::new();
     let owner = hir_body.id().hir_id.owner;
-    let bodies = &tcx.hir_owner_nodes(owner).unwrap().bodies;
-    hcx.with_hir_bodies(false, owner, bodies, |hcx| {
-        hir_body.value.hash_stable(hcx, &mut stable_hasher)
-    });
-    stable_hasher.finish()
+    tcx.hir_owner_nodes(owner).unwrap().hash_including_bodies.to_smaller_hash()
```

Comment:
@richkadel this pessimizes the hash used in coverage.  I could not find any documentation for this.  It is a big deal if the hash changes more often than strictly required?

Reply:
It's not well documented. [This document](https://llvm.org/docs/CoverageMappingFormat.html) says it's a "structural hash". I believe you want to make sure the hash does not change if the function source code does not change, and that the hash does change if the source changes.

What defines a change? Well, that's not clear. Since they call it a "structural hash", it _sounds_ to me like whitespace changes shouldn't count. But whitespace changes would affect lines and columns in coverage, so I'm not sure.

You said the hash may change "more often than strictly required". If by "more often" you mean it may sometimes change when it shouldn't, but usually it will be the same if the source doesn't change, then it's probably not a big deal.

Reply:
> You said the hash may change "more often than strictly required". If by "more often" you mean it may sometimes change when it shouldn't, but usually it will be the same if the source doesn't change, then it's probably not a big deal.

Thanks. I confirm that with this implementation, the hash will take into account spans and white space. Furthermore, the hash for a closure can change because of an unrelated modification in the same function. It hashes strictly more information.